pkg/controller/mizar/node_controller.go

```diff
@@ -0,0 +1,307 @@
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+)
+
+const (
+	NODE_KIND          string = "Node"
+	NODE_READY         string = "True"
+	NODE_CREATE        string = "Create"
+	NODE_UPDATE        string = "Update"
+	NODE_DELETE        string = "Delete"
+	NODE_RESUME        string = "Resume"
+	NODE_statusMessage string = "HANDLED"
+)
+
+type NodeController struct {
+	kubeclientset  *kubernetes.Clientset
+	informer       coreinformers.NodeInformer
+	informerSynced cache.InformerSynced
+	lister         corelisters.NodeLister
+	recorder       record.EventRecorder
+	queue          workqueue.RateLimitingInterface
+}
+
+func NewNodeController(kubeclientset *kubernetes.Clientset, nodeInformer coreinformers.NodeInformer) (*NodeController, error) {
+	informer := nodeInformer
+	eventBroadcaster := record.NewBroadcaster()
+	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "mizar-node-controller"})
+	eventBroadcaster.StartLogging(klog.Infof)
+	eventBroadcaster.StartRecordingToSink(
+		&v1core.EventSinkImpl{Interface: kubeclientset.CoreV1().Events("")})
+	queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+	c := &NodeController{
+		kubeclientset:  kubeclientset,
+		informer:       informer,
+		informerSynced: informer.Informer().HasSynced,
+		lister:         informer.Lister(),
+		recorder:       recorder,
+		queue:          queue,
+	}
+	klog.Infof(c.logInfoMessage("Sending events to api server"))
+	informer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(object interface{}) {
+			resource := object.(*v1.Node)
+			key := c.genKey(resource)
+			c.Enqueue(NODE_CREATE, key)
+			fmt.Printf("NODE_CREATE: %s", key)
+		},
+		UpdateFunc: func(oldObject, newObject interface{}) {
+			oldResource := oldObject.(*v1.Node)
+			newResource := newObject.(*v1.Node)
+			oldKey := c.genKey(oldResource)
+			newKey := c.genKey(newResource)
+			if oldKey == newKey {
+				klog.Infof(c.logEventMessage("No actual change in nodes, discarding: ", newKey))
+			} else {
+				command, err := c.compareKey(oldKey, newKey)
+				if err != nil {
+					klog.Errorf(c.logInfoMessage("Unexpected string in queue; discarding: " + newKey))
+				} else {
+					c.Enqueue(command, newKey)
+				}
+				fmt.Printf("NODE_UPDATE: %s, %s", command, newKey)
+			}
+		},
+		DeleteFunc: func(object interface{}) {
+			resource := object.(*v1.Node)
+			key := c.genKey(resource)
+			c.Enqueue(NODE_DELETE, key)
+			fmt.Printf("NODE_DELETE: %s", key)
+		},
+	})
+	return c, nil
+}
+
+// Run starts an asynchronous loop that detects events of cluster nodes.
+func (c *NodeController) Run(workers int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer c.queue.ShutDown()
+	klog.Infof(c.logInfoMessage("Starting node controller"))
+	klog.Infoln(c.logInfoMessage("Waiting cache to be synced"))
```

Comment:
nit: suggest to stick to Infof for consistence.

Reply:
All klog.Infoln() are changed to klog.infof()